### PR TITLE
[WFDISC-24] Add trace logging to discovery calls.

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR adds some tracing-level logging to each call to discovery.

For more details, see:  https://issues.jboss.org/browse/WFDISC-24

### CODE REVIEW

**Comment by rachmatowicz** [comment]
The log statements themselves might be a bit clumbsy...feedback appreciated. I needed an identifier to connect the two as the discovery may complete long after it was initiated, so I used the DiscoveryResult instance.

File: src/test/java/org/wildfly/discovery/ServiceDesignationTestCase.java [hunk 1]
```diff
@@ -50,7 +53,7 @@ public void testImpliesServiceTypeToServiceURL1() throws Exception {
     }
 
     @Test
-    public void testImpliesServiceTypeToServiceURL2() throws Exception {
+    public void testImpliesServiceTypeToServiceURL2() throws Exception {ServiceURL temp = new ServiceURL.Builder().setAbstractType("abstract").setAbstractTypeAuthority("ata").setUriSchemeAuthority("fred").setUri(URI.create("foo://bar.com:8080")).create();
```
Comment (dmlloyd) [inline src/test/java/org/wildfly/discovery/ServiceDesignationTestCase.java]: This formatting went weird.

**Comment by dmlloyd** [comment]
Other than one formatting thing, it's fine.

**Comment by rachmatowicz** [comment]
Removed the line from the test case that wasn't suppose to be there. Changed slightly the wording to make it more consistent.

**Pull request merged**